nh_clinical/patient.py

```diff
@@ -51,7 +51,34 @@ class NhClinicalPatient(osv.Model):
     ]
 
     @staticmethod
-    def _clean_identifiers(dirty_vals):
+    def _check_identifier_for_bad_chars(value):
+        """
+        Check for bad characters in string
+
+        :param value: string to check
+        """
+        unallowed_chars = '[^a-zA-Z0-9_\-\s]'
+        allowed_chars = re.sub(unallowed_chars, '', value)
+        if allowed_chars != value:
+            raise ValidationError(
+                'Patient identifier can only contain '
+                'alphanumeric characters, hyphens and underscores'
+            )
```

Comment:
@Nayira Can you confirm if use not borking on white space in the middle of a string and just removing this is acceptable?